organizers_bot/bot.py

```
@@ -87,6 +87,69 @@ async def create_challenge_channel(ctx: discord_slash.SlashContext,
         await ctx.send(f"The channel for <#{created.id}> ({category}) was created")
         await ctfnote.add_task(ctx, created, challenge, category, solved_prefix = "✓-", ctfid = ctfid)
 
+    @slash.slash(name="optout",
+                description="Opt-out of CTF participation.",
+                guild_ids=[config.bot.guild])
+    @require_role(config.mgmt.player_role)
+    async def opt_out(ctx: discord_slash.SlashContext):
+        await ctx.defer()
+
+        member = ctx.author
+        guild = ctx.guild
+
+        optout_role = discord.utils.get(guild.roles, id=config.mgmt.optout_player_role)
+        player_role = discord.utils.get(guild.roles, id=config.mgmt.player_role)
+
+        if not optout_role or not player_role:
+            await ctx.send("Error: roles are not configured correctly. Please contact an admin.", hidden=True)
+            return
+
+        if optout_role in member.roles:
+            await ctx.send("You are already opted out.", hidden=True)
+            return
+
+        try:
+            await member.add_roles(optout_role, reason=f"User opted out via /optout command")
+            if player_role in member.roles:
+                await member.remove_roles(player_role, reason=f"User opted out via /optout command")
+
+            await ctx.send(f"You have successfully opted out.")
+        except:
+            await ctx.send("Error: I don't have permissions? to manage roles. Please contact an admin.", hidden=True)
```

maybe re-raise the error afterwards, so at least we have some sort of log of what actually went wrong

---

Got rid of error handling like the rest of the code 😅